tokenizer_test.go

```diff
@@ -472,3 +493,250 @@ func BenchmarkDecodeNTokens(b *testing.B) {
+			wantError: true,
+			checkDir:  nil, // No directory check needed for error case
+		},
+		{
+			name:    "nonexistent model",
+			modelID: "nonexistent/model",
+			setupOpts: func() []tokenizers.TokenizerConfigOption {
+				tmpDir := t.TempDir()
+				return []tokenizers.TokenizerConfigOption{
+					tokenizers.WithCacheDir(tmpDir),
+				}
+			},
+			wantError: true,
+			checkDir:  nil, // No directory check needed for error case
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			origTransport := http.DefaultClient.Transport
+			http.DefaultClient.Transport = &mockTransport{
+				server:  server,
+				modelID: tt.modelID,
+			}
+			defer func() { http.DefaultClient.Transport = origTransport }()
+
+			tokenizer, err := tokenizers.FromPretrained(tt.modelID, tt.setupOpts()...)
+
+			if tt.wantError {
+				if err == nil {
+					t.Error("expected error, got nil")
+				}
+				return
+			}
+
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+				return
+			}
+
+			if tokenizer == nil {
+				t.Error("expected tokenizer, got nil")
+				return
+			}
+
+			if err := tokenizer.Close(); err != nil {
+				t.Errorf("error closing tokenizer: %v", err)
+			}
+		})
+	}
+}
+
+func TestConfigOptions(t *testing.T) {
+	server := mockHuggingFaceServer(t)
+	defer server.Close()
+
+	t.Run("WithCacheDir", func(t *testing.T) {
+		tmpDir := t.TempDir()
+
+		origTransport := http.DefaultClient.Transport
+		http.DefaultClient.Transport = &mockTransport{
+			server:  server,
+			modelID: "bert-base-uncased",
+		}
+		defer func() { http.DefaultClient.Transport = origTransport }()
+
+		tokenizer, err := tokenizers.FromPretrained(
+			"bert-base-uncased",
+			tokenizers.WithCacheDir(tmpDir),
+		)
+
+		if err != nil {
+			t.Errorf("unexpected error: %v", err)
+			return
+		}
+		defer tokenizer.Close()
+
+		files := []string{"tokenizer.json", "vocab.txt", "special_tokens_map.json"}
+		for _, file := range files {
+			path := filepath.Join(tmpDir, "bert-base-uncased", file)
+			if _, err := os.Stat(path); os.IsNotExist(err) {
+				t.Errorf("expected file %s to exist", file)
+			}
+		}
+	})
+
+	t.Run("WithAuthToken", func(t *testing.T) {
+		origTransport := http.DefaultClient.Transport
+		http.DefaultClient.Transport = &mockTransport{
+			server:  server,
+			modelID: "private/model",
+		}
+		defer func() { http.DefaultClient.Transport = origTransport }()
+
+		tokenizer, err := tokenizers.FromPretrained(
+			"private/model",
+			tokenizers.WithAuthToken("test-token"),
+			tokenizers.WithCacheDir(t.TempDir()),
+		)
+
+		if err != nil {
+			t.Errorf("unexpected error with valid auth token: %v", err)
+			return
+		}
+		if tokenizer != nil {
+			tokenizer.Close()
+		}
+
+		tokenizer, err = tokenizers.FromPretrained(
+			"private/model",
+			tokenizers.WithAuthToken("invalid-token"),
+			tokenizers.WithCacheDir(t.TempDir()),
+		)
+
+		if err == nil {
+			t.Error("expected error with invalid auth token, got nil")
```

Comment:
rewrite these in table test format, where expected error, token are parameters, but the test body is the same

Reply:
https://go.dev/wiki/TableDrivenTests